contracts/VaultV3.vy

```diff
@@ -783,26 +803,22 @@ def _process_report(strategy: address) -> (uint256, uint256):
     # Protocol fee assessment
     protocol_fees: uint256 = 0
     protocol_fee_recipient: address = empty(address)
-    seconds_since_last_report: uint256 = block.timestamp - self.last_report
-    # to avoid wasting gas for minimal fees vault will only assess once every PROTOCOL_FEE_ASSESSMENT_PERIOD seconds
-    if(seconds_since_last_report >= PROTOCOL_FEE_ASSESSMENT_PERIOD):
-      protocol_fee_bps: uint16 = 0
-      protocol_fee_last_change: uint32 = 0
-
-      protocol_fee_bps, protocol_fee_last_change, protocol_fee_recipient = IFactory(FACTORY).protocol_fee_config()
+    protocol_fees, protocol_fee_recipient = self._assess_protocol_fees()
+    total_fees += protocol_fees
 
-      if(protocol_fee_bps > 0):
-        # NOTE: charge fees since last report OR last fee change (this will mean less fees are charged after a change in protocol_fees, but fees should not change frequently)
-        seconds_since_last_report = min(seconds_since_last_report, block.timestamp - convert(protocol_fee_last_change, uint256))
-        protocol_fees = convert(protocol_fee_bps, uint256) * self._total_assets() * seconds_since_last_report / 24 / 365 / 3600 / MAX_BPS
-        total_fees += protocol_fees
-        self.last_report = block.timestamp
-   
     # We calculate the amount of shares that could be insta unlocked to avoid pps changes
     # NOTE: this needs to be done before any pps changes
-    shares_to_burn: uint256 = 0 
+    shares_to_burn: uint256 = 0
+    fees_shares: uint256 = 0
+    protocol_fees_shares: uint256 = 0
     if loss + total_fees > 0:
-      shares_to_burn = self._convert_to_shares(loss + total_fees)
+        shares_to_burn += self._convert_to_shares(loss + total_fees)
+        # Vault calculates the amount of shares to mint as fees before changing totalAssets / totalSupply
+        if total_fees > 0:
+            fees_shares = self._convert_to_shares(total_fees)
+            if protocol_fees > 0:
+              protocol_fees_shares = self._convert_to_shares(protocol_fees)
```

Comment:
So this will be slightly inaccurate when there is an overall loss right? Because if we have total loss that cant be accounted for with the burn pps will decrease. Meaning we should actually be issuing more shares for the fee collectors than what this will return since the total_debt has yet to be updated.

This shows up in the test_loss_fees_refunds test in test_profit_ulocking

Reply:
I think you could just move it below the burn shares part and it should be good,

calculating shares_to_burn there should be good since in the case of pps decrease it will burn the previously_locked_shares + newly_locked in the min check right?